File: frontend/src/api/usage.ts
import { apiGet } from "./client";

export interface AIUsageFeatureCount {
  feature_type: string;
  count: number;
  success_count: number;
  failure_count: number;
}

export interface AIUsageLog {
  id: number;
  feature_type: string;
  model_name: string;
  status: "success" | "failed";
  error_message: string | null;
  input_tokens: number | null;
  output_tokens: number | null;
  total_tokens: number | null;
  estimated_cost: number | null;
  created_at: string;
}

export interface AIUsageSummary {
  monthly_quota: number;
  monthly_used: number;
  monthly_remaining: number;
  total_call_count: number;
  monthly_success_count: number;
  monthly_failure_count: number;
  feature_counts: AIUsageFeatureCount[];
  recent_calls: AIUsageLog[];
}

export function getUsageSummary(): Promise<AIUsageSummary> {
  return apiGet<AIUsageSummary>("/usage/summary");
}

File: backend/app/schemas/usage.py
from datetime import datetime

from pydantic import BaseModel


class AIUsageFeatureCountRead(BaseModel):
    feature_type: str
    count: int
    success_count: int
    failure_count: int


class AIUsageLogRead(BaseModel):
    id: int
    feature_type: str
    model_name: str
    status: str
    error_message: str | None
    input_tokens: int | None
    output_tokens: int | None
    total_tokens: int | None
    estimated_cost: float | None
    created_at: datetime


class AIUsageSummaryRead(BaseModel):
    monthly_quota: int
    monthly_used: int
    monthly_remaining: int
    total_call_count: int
    monthly_success_count: int
    monthly_failure_count: int
    feature_counts: list[AIUsageFeatureCountRead]
    recent_calls: list[AIUsageLogRead]
